### ISSUE TITLE
Enable InstallFailureSignalHandler for Caffe Windows and upgrade glog version

### ISSUE THREAD

**Issue by dashesy** [BVLC/caffe#5926]
### Issue summary

glog did not have `InstallFailureSignalHandler` for Windows, which resulted in empty stack traces after `*** Check failure stack trace: **` on Windows, but after [this commit](https://github.com/google/glog/commit/d1f49ba5aa8a06be74dedc60cdf56a2e74031b7b#diff-c6c90d642eb73bc10e4b32bd8a4588c0) glog now exports that symbol.

### Steps to reproduce

Run training with an invalid path in `solver.prototxt` there will be a crash, but without any context as to why.

After updating glog to the latest version [this line](https://github.com/BVLC/caffe/blob/windows/src/caffe/common.cpp#L57) can be enabled for Windows too (remove `#ifdef` around it), hopefully for better debugging messages.

### Your system configuration
Operating system: Windows
Compiler: Visual Studio
CUDA version (if applicable): 8.0
CUDNN version (if applicable): 5.1
BLAS: MKL
Python or MATLAB version (for pycaffe and matcaffe respectively): Python 3.5

**Comment by willyd** [comment]
@dashesy Thanks for reporting.

Am I correct that this commit is not included in any official release? 

If this is the case I suggest that you submit a small patch to the [vcpkg](https://github.com/Microsoft/vcpkg) port of glog. I am currently working to port the dependency management of caffe to vcpkg. So if this change is in when I am done with the port this will resolve the issue. Thanks.

**Comment by dashesy** [comment]
Yes it looks like it is not included in any official release yet.
The current version is 0.3.5. I think 0.3.6 will have the fix, then I will submit a patch to `vcpkg` port of glog to bump the version.

**Comment by zchrissirhcz** [comment]
@willyd @dashesy How is this glog issue going on? I'm on windows and trying to build Caffe-SSD with cmake and encounter this problem. Any suggestion? Thanks.
